### PR TITLE
Update across the board ownership changes, url updates

### PR DESCRIPTION
**Pull request by iainnash**
Fix URL changes, ownership updates.

Allow appending delete add operations for users to the current transaction queue.

### CODE REVIEW

File: src/hooks/useSetParamsFromQuery.ts [hunk 1]
```diff
@@ -11,11 +11,15 @@ export const useRedirectToProposalWithNewParams = () => {
       if (!proposal.actions?.length) {
         return;
       }
-      console.log("setting params", proposal.actions);
       setParams({
         [queryKeys.targets]: proposal.actions!.map((action) => action.to).join("|"),
         [queryKeys.calldatas]: proposal.actions!.map((action) => action.data).join("|"),
         [queryKeys.values]: proposal.actions!.map((action) => action.value).join("|"),
+        ...(proposal.nonce
```
Comment (oveddan) [inline src/hooks/useSetParamsFromQuery.ts:18]: great to have a nonce in here!

File: src/hooks/useLoadProposalFromQuery.ts [hunk 2]
```diff
@@ -5,17 +5,20 @@
...
   targets: "targets",
   calldatas: "calldatas",
-  values: "values"
-};
+  values: "values",
+  nonce: "nonce",
+} as const;
 
 export const useLoadProposalFromQuery = () => {
   const [proposal, setProposal] = useState<undefined | Proposal>();
```
Comment (oveddan) [inline src/hooks/useLoadProposalFromQuery.ts:13]: were going to have a state sync issue here - proposal is now saved here, and in another place where the context is.

File: src/components/SetOwnerModal.tsx [hunk 3]
```diff
@@ -76,7 +74,7 @@ const AddOwnerModalContent = ({ onClose }: { onClose: () => void }) => {
       }}
     >
       <Form>
-        <Text variant="featured-2">Add Owner</Text>
+        <Text variant="featured-2">Add Signer</Text>
```
Comment (IsabellaSmallcombe) [inline src/components/SetOwnerModal.tsx:77]: Should the other variables be changed to signer too? noticed they all still are called owner

File: src/app/NewSafeProposal.tsx [hunk 4]
```diff
@@ -428,22 +436,24 @@ export const NewSafeProposal = () => {
...
-        )}
-        {!isEditing && proposal && (
-          <ViewProposal
-            proposal={proposal}
-            handleEditClicked={handleEditClicked}
-          />
-        )}
-      </SafeInformation>
-    </View>
+    <ProposalContext.Provider value={proposal}>
```
[the remark this answers is not in the document]
Comment (oveddan) [reply src/app/NewSafeProposal.tsx:439]: did this change in a follow on PR #27

**Comment by iainnash** [comment]
Do you think this issue would be alleviated if we use the context
everywhere and remove the other state you suggested? would increase perf
and speed up the component tree. another cool benefit is we could auto
update the url from set state through a useUpdateUrl effect.

**Comment by oveddan** [comment]
I tend to avoid using context unless absolutely necessary - its tricky to track state changes and it is a mutable state, which often doesnt cause components to re-render.  

Explicitly passing down props has more predictable results.  

Since we want to be able to pass the url around, we should just make that the source of truth, and update that accordingly; this way the state will be reflected in the current url, and we shouldn't need to duplicate that in a context

> Do you think this issue would be alleviated if we use the context
> everywhere and remove the other state you suggested? would increase perf
> and speed up the component tree. another cool benefit is we could auto
> update the url from set state through a useUpdateUrl effect.
>

**Comment by oveddan** [comment]
### Merge activity

* **Jun 21, 3:47 PM EDT**: @oveddan started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/ourzora/smol-safe/26).

**Pull request merged**